packages/sitecore-jss/src/layout-service.ts

```diff
@@ -228,13 +236,15 @@ export class GraphQLLayoutService implements LayoutService {
    * @returns {Promise<LayoutServiceData>} layout service data
    */
   async fetchLayoutData(itemPath: string, language?: string): Promise<LayoutServiceData> {
-    const query = this.getLayoutQuery(itemPath, language);
+    const query = this.serviceConfig.formatSearchQuery
+      ? this.serviceConfig.formatSearchQuery(this.serviceConfig.siteName, itemPath, language)
+      : this.getLayoutQuery(itemPath, language);
```

Comment:
@illiakovalenko This works differently than the sitemap service, where the `formatSearchQuery` is only responsible for the query arguments fragment (not the entire query). Should this work the same way to be consistent?

Reply:
@ambrauer I wanted to do in same way, but we have common LayoutService interface, and I can't provide formatSearchQuery there

Reply:
@ambrauer In this case we provide ability to fully override query, in case of `sitemap-service` we added ability to override ONLY search conditions

Reply:
@illiakovalenko I guess that's the concern. Both options are named "formatSearchQuery", but expect different string fragments.

Reply:
May be in same way I user should have ability to change only arguments, instead of response